Source/Core/Core/PowerPC/JitArm64/JitAsm.cpp

```diff
@@ -287,19 +287,18 @@ void JitArm64::GenerateFres()
   CMP(ARM64Reg::X2, 895);
   FixupBranch small_exponent = B(CCFlags::CC_LO);
 
-  MOVI2R(ARM64Reg::X4, 1148LL);
-  CMP(ARM64Reg::X2, ARM64Reg::X4);
+  CMP(ARM64Reg::X2, 1148);
   FixupBranch large_exponent = B(CCFlags::CC_HI);
 
   UBFX(ARM64Reg::X2, ARM64Reg::X1, 47, 5);  // Grab upper part of mantissa
   MOVP2R(ARM64Reg::X3, &Common::fres_expected);
   ADD(ARM64Reg::X2, ARM64Reg::X3, ARM64Reg::X2, ArithOption(ARM64Reg::X2, ShiftType::LSL, 3));
-  LDP(IndexType::Signed, ARM64Reg::W2, ARM64Reg::W3, ARM64Reg::X2, 0);
   UBFX(ARM64Reg::X1, ARM64Reg::X1, 37, 10);  // Grab lower part of mantissa
+  LDP(IndexType::Signed, ARM64Reg::W2, ARM64Reg::W3, ARM64Reg::X2, 0);
```

Comment:
I'm unfamiliar with the exact ARM in-order semantics, but can the `MOVI2R(ARM64Reg::W4, 1)` here be moved further up? Would that change anything?

Reply:
Yes, we could move it as far up as we want to as long as we don't move it before an instruction that uses W4/X4 (which there are none of here). But there's no scheduling benefit to doing so. This MOV can run in parallel with the LDP, assuming the CPU is capable of running instructions in parallel at all, which we kind of have to assume when deciding on scheduling since otherwise it doesn't matter how we schedule instructions.

Reply:
Gotcha, thanks for the explanation.